[BUG] Bokeh 2.3 properties are non-nullable by default

ref: http://docs.bokeh.org/en/latest/docs/releases.html#properties-are-non-nullable-by-default

Custom models will require a small update to function:
```
Compilation failed:

../../../../anaconda/envs/rapids-0.18/lib/python3.8/site-packages/cuxfilter/charts/datashader/custom_extensions/graph_inspect_widget.py:CustomInspectTool.ts:12:11 - error TS2322: Type 'boolean | undefined' is not assignable to type 'Boolean'.
  Type 'undefined' is not assignable to type 'Boolean'.

12           this.model._active = this.model.properties.active.spec.value
             ~~~~~~~~~~~~~~~~~~
../../../../anaconda/envs/rapids-0.18/lib/python3.8/site-packages/cuxfilter/charts/datashader/custom_extensions/graph_inspect_widget.py:CustomInspectTool.ts:12:61 - error TS2445: Property 'spec' is protected and only accessible within class 'Property<T>' and its subclasses.

12           this.model._active = this.model.properties.active.spec.value
                                                               ~~~~
ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.

```

cc @AjayThorve

--- a/ci/gpu/build.sh
+++ b/ci/gpu/build.sh
@@ -59,13 +59,13 @@ gpuci_conda_retry install -y \
                "cuspatial=$MINOR_VERSION.*" \
                "dask-cudf=$MINOR_VERSION.*" "dask-cuda=$MINOR_VERSION.*" \
                "numba>=0.51.2" \
-               "bokeh>=2.3.2,<2.4" \
+               "bokeh<=2.2.3" \
                "rapids-build-env=$MINOR_VERSION.*" \
                "rapids-notebook-env=$MINOR_VERSION.*"
 
 # https://docs.rapids.ai/maintainers/depmgmt/ 
-# conda remove -f rapids-build-env rapids-notebook-env
-# conda install "your-pkg=1.0.0"
+conda remove -f rapids-build-env rapids-notebook-env
+conda install "bokeh>=2.3.2, <2.4"
 
 gpuci_logger "Check versions"
 python --version

--- a/ci/gpu/build.sh
+++ b/ci/gpu/build.sh
@@ -59,7 +59,7 @@ gpuci_conda_retry install -y \
                "cuspatial=$MINOR_VERSION.*" \
                "dask-cudf=$MINOR_VERSION.*" "dask-cuda=$MINOR_VERSION.*" \
                "numba>=0.51.2" \
-               "bokeh<=2.2.3" \
+               "bokeh>=2.3.*,<2.4.*" \
                "rapids-build-env=$MINOR_VERSION.*" \
                "rapids-notebook-env=$MINOR_VERSION.*"
 

--- a/ci/gpu/build.sh
+++ b/ci/gpu/build.sh
@@ -64,7 +64,7 @@ gpuci_conda_retry install -y \
                "rapids-notebook-env=$MINOR_VERSION.*"
 
 # https://docs.rapids.ai/maintainers/depmgmt/ 
-# conda remove -f rapids-build-env rapids-notebook-env
+# conda remove --force rapids-build-env rapids-notebook-env
 # conda install "your-pkg=1.0.0"
 
 gpuci_logger "Check versions"

--- a/ci/gpu/build.sh
+++ b/ci/gpu/build.sh
@@ -59,12 +59,13 @@ gpuci_conda_retry install -y \
                "cuspatial=$MINOR_VERSION.*" \
                "dask-cudf=$MINOR_VERSION.*" "dask-cuda=$MINOR_VERSION.*" \
                "numba>=0.51.2" \
+               "bokeh>=2.3.2,<2.4" \
                "rapids-build-env=$MINOR_VERSION.*" \
                "rapids-notebook-env=$MINOR_VERSION.*"
 
 # https://docs.rapids.ai/maintainers/depmgmt/ 
-conda remove --force rapids-build-env rapids-notebook-env
-conda install "bokeh>=2.3.2, <2.4"
+# conda remove -f rapids-build-env rapids-notebook-env
+# conda install "your-pkg=1.0.0"
 
 gpuci_logger "Check versions"
 python --version

--- a/ci/gpu/build.sh
+++ b/ci/gpu/build.sh
@@ -59,7 +59,6 @@ gpuci_conda_retry install -y \
                "cuspatial=$MINOR_VERSION.*" \
                "dask-cudf=$MINOR_VERSION.*" "dask-cuda=$MINOR_VERSION.*" \
                "numba>=0.51.2" \
-               "bokeh<=2.2.3" \
                "rapids-build-env=$MINOR_VERSION.*" \
                "rapids-notebook-env=$MINOR_VERSION.*"
 

--- a/ci/gpu/build.sh
+++ b/ci/gpu/build.sh
@@ -59,7 +59,7 @@ gpuci_conda_retry install -y \
                "cuspatial=$MINOR_VERSION.*" \
                "dask-cudf=$MINOR_VERSION.*" "dask-cuda=$MINOR_VERSION.*" \
                "numba>=0.51.2" \
-               "bokeh>=2.3.*,<2.4.*" \
+               "bokeh>=2.3.2,<2.4" \
                "rapids-build-env=$MINOR_VERSION.*" \
                "rapids-notebook-env=$MINOR_VERSION.*"
 

--- a/ci/gpu/build.sh
+++ b/ci/gpu/build.sh
@@ -63,7 +63,7 @@ gpuci_conda_retry install -y \
                "rapids-notebook-env=$MINOR_VERSION.*"
 
 # https://docs.rapids.ai/maintainers/depmgmt/ 
-conda remove -f rapids-build-env rapids-notebook-env
+conda remove --force rapids-build-env rapids-notebook-env
 conda install "bokeh>=2.3.2, <2.4"
 
 gpuci_logger "Check versions"

--- a/conda/environments/cuxfilter_dev_cuda10.1.yml
+++ b/conda/environments/cuxfilter_dev_cuda10.1.yml
@@ -20,7 +20,7 @@ dependencies:
 - pyproj>=2.4, <=3.1
 - libwebp
 - pandoc=<2.0.0
-- bokeh>=2.1.1,<=2.2.3
+- bokeh>=2.3.*,<2.4.*
 - panel>=0.10.3
 - cudatoolkit=10.1
 - nodejs

--- a/conda/environments/cuxfilter_dev_cuda10.1.yml
+++ b/conda/environments/cuxfilter_dev_cuda10.1.yml
@@ -20,7 +20,7 @@ dependencies:
 - pyproj>=2.4, <=3.1
 - libwebp
 - pandoc=<2.0.0
-- bokeh>=2.3.*,<2.4.*
+- bokeh>=2.3.2,<2.4
 - panel>=0.10.3
 - cudatoolkit=10.1
 - nodejs

--- a/conda/environments/cuxfilter_dev_cuda10.2.yml
+++ b/conda/environments/cuxfilter_dev_cuda10.2.yml
@@ -21,7 +21,7 @@ dependencies:
 - pyproj>=2.4, <=3.1
 - libwebp
 - pandoc=<2.0.0
-- bokeh>=2.1.1,<=2.2.3
+- bokeh>=2.3.*,<2.4.*
 - panel>=0.10.3
 - cudatoolkit=10.2
 - nodejs

--- a/conda/environments/cuxfilter_dev_cuda10.2.yml
+++ b/conda/environments/cuxfilter_dev_cuda10.2.yml
@@ -21,7 +21,7 @@ dependencies:
 - pyproj>=2.4, <=3.1
 - libwebp
 - pandoc=<2.0.0
-- bokeh>=2.3.*,<2.4.*
+- bokeh>=2.3.2,<2.4
 - panel>=0.10.3
 - cudatoolkit=10.2
 - nodejs

--- a/conda/environments/cuxfilter_dev_cuda11.0.yml
+++ b/conda/environments/cuxfilter_dev_cuda11.0.yml
@@ -20,7 +20,7 @@ dependencies:
 - pyproj>=2.4, <=3.1
 - libwebp
 - pandoc=<2.0.0
-- bokeh>=2.1.1,<=2.2.3
+- bokeh>=2.3.*,<2.4.*
 - panel>=0.10.3
 - cudatoolkit=11.0
 - nodejs

--- a/conda/environments/cuxfilter_dev_cuda11.0.yml
+++ b/conda/environments/cuxfilter_dev_cuda11.0.yml
@@ -20,7 +20,7 @@ dependencies:
 - pyproj>=2.4, <=3.1
 - libwebp
 - pandoc=<2.0.0
-- bokeh>=2.3.*,<2.4.*
+- bokeh>=2.3.2,<2.4
 - panel>=0.10.3
 - cudatoolkit=11.0
 - nodejs

--- a/conda/recipes/cuxfilter/meta.yaml
+++ b/conda/recipes/cuxfilter/meta.yaml
@@ -35,7 +35,7 @@ requirements:
     - numba >=0.51.2
     - cupy >=7.8.0,<10.0.0a0
     - panel >=0.10.3
-    - bokeh >=2.1.1,<=2.2.3
+    - bokeh >=2.3.*,<2.4.*
     - pyproj >=2.4, <=3.1
     - geopandas >=0.9.0,<0.10.0a0
     - nodejs >=12,<15

--- a/conda/recipes/cuxfilter/meta.yaml
+++ b/conda/recipes/cuxfilter/meta.yaml
@@ -35,7 +35,7 @@ requirements:
     - numba >=0.51.2
     - cupy >=7.8.0,<10.0.0a0
     - panel >=0.10.3
-    - bokeh >=2.3.*,<2.4.*
+    - bokeh >=2.3.2,<2.4
     - pyproj >=2.4, <=3.1
     - geopandas >=0.9.0,<0.10.0a0
     - nodejs >=12,<15

--- a/python/cuxfilter/charts/datashader/custom_extensions/graph_inspect_widget.py
+++ b/python/cuxfilter/charts/datashader/custom_extensions/graph_inspect_widget.py
@@ -1,4 +1,4 @@
-from bokeh.core.properties import Image, Bool, String
+from bokeh.core.properties import Image, Bool, String, Nullable
 from bokeh.models import Tool
 from bokeh.util.compiler import TypeScript
 
@@ -13,7 +13,7 @@ export class CustomInspectToolView extends InspectToolView {
       super.connect_signals()
 
       this.connect(this.model.properties.active.change, () => {
-          this.model._active = this.model.properties.active.spec.value
+          this.model._active = this.model.active
       })
   }
 }
@@ -53,6 +53,6 @@ export class CustomInspectTool extends InspectTool {
 
 class CustomInspectTool(Tool):
     __implementation__ = TypeScript(TS_CODE)
-    _active = Bool
+    _active = Nullable(Bool)
     icon = Image()
-    tool_name = String
+    tool_name = String()

--- a/python/cuxfilter/layouts/custom_react_template.py
+++ b/python/cuxfilter/layouts/custom_react_template.py
@@ -1,9 +1,5 @@
-from urllib.parse import urljoin
+from panel import depends
 import param
-from panel.io.resources import LOCAL_DIST
-from panel.io.state import state
-from panel.util import url_path
-from panel.depends import depends
 from panel.layout import Card, GridSpec
 from panel.template.base import BasicTemplate
 import os
@@ -70,30 +66,18 @@ class ReactTemplate(BasicTemplate):
     }
 
     def _template_resources(self):
-        name = type(self).__name__.lower()
-        base_url = state.base_url
-        if state.base_url.startswith("/"):
-            base_url = state.base_url[1:]
-        dist_path = urljoin(base_url, LOCAL_DIST)
-        # External resources
-        css_files = dict(self._resources["css"])
-        for cssname, css in css_files.items():
-            css_path = url_path(css)
-            css_files[cssname] = dist_path + f"bundled/{name}/{css_path}"
-        js_files = dict(self._resources["js"])
-        for jsname, js in js_files.items():
-            js_path = url_path(js)
-            js_files[jsname] = dist_path + f"bundled/{name}/{js_path}"
-
+        resources = super()._template_resources()
         # CSS files
         base_css = os.path.basename(self._css)
-        css_files["base"] = f"{CUSTOM_DIST_PATH_LAYOUTS}/{base_css}"
+        resources["css"]["base"] = f"{CUSTOM_DIST_PATH_LAYOUTS}/{base_css}"
         if self.theme:
             theme = self.theme.find_theme(type(self))
             if theme and theme.css:
                 basename = os.path.basename(theme.css)
-                css_files["theme"] = f"{CUSTOM_DIST_PATH_THEMES}/{basename}"
-        return {"css": css_files, "js": js_files}
+                resources["css"]["theme"] = (
+                    f"{CUSTOM_DIST_PATH_THEMES}/{basename}"
+                )
+        return resources
 
     def __init__(self, **params):
         if "main" not in params:

--- a/python/cuxfilter/layouts/custom_react_template.py
+++ b/python/cuxfilter/layouts/custom_react_template.py
@@ -74,9 +74,9 @@ class ReactTemplate(BasicTemplate):
             theme = self.theme.find_theme(type(self))
             if theme and theme.css:
                 basename = os.path.basename(theme.css)
-                resources["css"]["theme"] = (
-                    f"{CUSTOM_DIST_PATH_THEMES}/{basename}"
-                )
+                resources["css"][
+                    "theme"
+                ] = f"{CUSTOM_DIST_PATH_THEMES}/{basename}"
         return resources
 
     def __init__(self, **params):

